dynamics/motor_efficiency/motor_efficiency.py

```diff
@@ -0,0 +1,118 @@
+import argparse
+import math
+import matplotlib.pyplot as plt
+import pandas as pd
+from sklearn import linear_model
+from scipy.optimize import curve_fit
+import numpy as np
+
+def log_func(self, x, a, b):
+    return a + b * np.log(x)
+
+def linear_func(self, x, a, b):
+    return a + b * x
+
+def exp_func(self, x, a, b, c):
+    return a * np.exp(b * x) + c
+
+class MotorEfficiency:
+
+    speed_torque_func = log_func
+
+    def __init__(self, coil):
+        self.coil = coil
+        self.dataset = pd.read_csv(coil + 'Data.csv')
+
+        if coil == "HI":
+            # remove outlier that exists in hi coil data 
+            self.dataset.drop(axis=0, index=11, inplace=True)
+
+        # let us determine relationship between speed and torque with voltage*current
+        # we will use this predicted value in place of actual current and voltage
+        x = self.dataset[['Speed', 'Torque']]
+        y = self.dataset['Voltage'] * self.dataset['Current']
+        regr = linear_model.LinearRegression()
+        regr.fit(x, y)
+
+        self.speed_k = regr.coef_[0]
+        self.torque_k = regr.coef_[1]
+        self.c = regr.intercept_
+
+        # generate fit of speed and torque points to generate data
+        # see speed_torque_func for function used
+        self.rotational_speed = self.dataset['Speed'].values
+        self.torque = self.dataset['Torque'].values
+
+        self.start_pars, cov = curve_fit(f=self.speed_torque_func, xdata=self.rotational_speed, ydata=self.torque)
```

Comment:
where does cov get used, it doesn't seem to be an attribute for the class.

Reply:
Oh nope, I'll just make it unnamed